cmd/docker-mcp/server/ls.go

```
@@ -18,5 +41,257 @@ func List(ctx context.Context, docker docker.Client) ([]string, error) {
 		return nil, err
 	}
 
-	return registry.ServerNames(), nil
+	// Read user's configuration to populate registry tiles
+	userConfigYAML, err := config.ReadConfig(ctx, docker)
+	if err != nil {
+		return nil, fmt.Errorf("reading user config: %w", err)
+	}
+
+	userConfig, err := config.ParseConfig(userConfigYAML)
+	if err != nil {
+		return nil, fmt.Errorf("parsing user config: %w", err)
+	}
+
+	// Populate registry tiles with user config
+	for serverName, tile := range registry.Servers {
+		if len(tile.Config) == 0 {
+			if userServerConfig, hasUserConfig := userConfig[serverName]; hasUserConfig {
+				tile.Config = userServerConfig
+				registry.Servers[serverName] = tile
+			}
+		}
+	}
+
+	// Read the catalog to get server metadata (descriptions, etc.)
+	catalogData, err := catalog.Get(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the list of configured secrets
+	configuredSecrets, err := desktop.NewSecretsClient().ListJfsSecrets(ctx)
+	if err != nil {
+		// If we can't get secrets, assume none are configured
+		if !quiet {
+			fmt.Fprintf(os.Stderr, "Warning: error fetching secrets: %v\n", err)
+		}
```

Should we return `err` here?

---

Ah I think I see what you mean, secrets and OAuth just silently fail and log the error to the terminal ( if not in JSON format )
Not sure how hard of an error we want for them, specially OAuth if it fails to fetch for X or Y reasons

---

@saucow Currently for getting the OAuth and Secrets we silently fail and print out a warning. A lot of the FE relies on the base return of the server names, since I'm not super well versed on how getting the OAuth and Secrets behave... is this overkill? should I instead just break the command? or are they likely to fail for external reasons?